src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -321,5 +322,10 @@ public override string GetNodeCustomName(object smoObject, SmoQueryContext smoCo
         {
             return string.Empty;
         }
+
+        public override string GetNodePathName(object smoObject)
+        {
+            return (smoObject as NamedSmoObject).Name;

```

Comment:
I'm not sure if all SMO objects are NamedSmoObject, do you have to check for null here?